sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/LiveMetricsLogProcessor.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Monitor.OpenTelemetry.LiveMetrics.Internals;
+using Azure.Monitor.OpenTelemetry.LiveMetrics.Internals.DataCollection;
+using OpenTelemetry;
+using OpenTelemetry.Logs;
+
+namespace Azure.Monitor.OpenTelemetry.LiveMetrics
+{
+    internal class LiveMetricsLogProcessor : BaseProcessor<LogRecord>
+    {
+        private bool _disposed;
+        private LiveMetricsResource? _resource;
+        private readonly Manager _manager;
+
+        internal LiveMetricsResource? LiveMetricsResource => _resource ??= ParentProvider?.GetResource().CreateAzureMonitorResource();
+
+        public LiveMetricsLogProcessor(Manager manager)
+        {
+            _manager = manager;
+        }
+
+        public override void OnEnd(LogRecord data)
+        {
+            // Check if live metrics is enabled.
+            if (!_manager.ShouldCollect())
+            {
+                return;
+            }
+
+            // Resource is not available at initialization and must be set later.
+            if (_manager.LiveMetricsResource == null && LiveMetricsResource != null)
+            {
+                _manager.LiveMetricsResource = LiveMetricsResource;
+            }
+
+            if (data.Exception is null)
+            {
+                AddLogDocument(data);
+            }
+            else
+            {
+                AddExceptionDocument(data.Exception);
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    try
+                    {
+                        _manager.Dispose();
```

Comment:
If manager is shared for the same `options`, what happens to others which are using this manager if you dispose one of them?